### ISSUE TITLE
Song randomly ends

### ISSUE THREAD

**Issue by C0rn3j** [ampache/ampache#1472]
# Environment

 * Ampache version: 3.8.2 AND 3.8.2.223.g18aeae2a-1(which seems to be 3.8.3-develop)
 * Web server + version: nginx-mainline 1.11.8
 * Server and Client operating system: Arch Linux

 * Client type:
   - Web

# Settings
What I got when I configured it + enabling debug. Can post the file if relevant.

# Description
One of the songs ends at 0:23 instead of 3:xx.
Happens both in firefox and chromium.

On the dev version I can't play a lot of the songs at all.

# Logs
3.8.2 http://pastebin.com/Xy6Bi1xA

**Comment by C0rn3j** [comment]
Managed to get the git devel version running properly, same issue as with the stable one.

**Comment by kiv57** [comment]
Not sure if it's the same issue :
In web UI, playback of lossless files (flac, m4a) often ends in the middle of the song (not exactly the middle!).

Config :
=====
- Debian 8
- Ampache 3.8.3-develop
- nginx 1.6.2
- php  5.6.24-0+deb8u1

What I already tried with no effect :
=====
- enabled html5 only
- enabled flash only
- changed in config file the line to `encode_args_mp3 = "-vn -b:a %SAMPLE%K -c:a libmp3lame -f mp3 pipe:1"`
- disabled ogg transcoding in config file.
- Firefox (W7), Chrome (W7), Safari (OS X) give same issue

Don't know where to investigate... any help appreciated!

**Comment by C0rn3j** [comment]
It likely is, so far it only happened with FLACs to me.

**Comment by wagnered** [comment]
@C0rn3j You and @kiv57 are both using NGinx as web server.  I'm wondering if the problem is in you server settings.  I am using lighttpd and have just run 6 or 7 flac audios some as long as 7 minutes without any premature song endings.

Also, when using the web player, You should always see the gray color filling up the buffer as the blue starts indicating the playing of the song.  The gray should always fill up the progression line.  If not, it might be further indications that the complete file not being transferred becasue of something going on with NGinx.

**Comment by C0rn3j** [comment]
@wagnered I've sent one of the files that breaks it to the email listed on your github account, feel free to test

**Comment by wagnered** [comment]
Interesting, @Martin,  I added the song to a separate catalog and it
doesn't even load using the web-player. It displays in the playlist for
less than a second then disappears.

It does play the full length locally with Audicious, though.    Was this
ripped from the video? By you? if so, what did you use to rip it- ffmpeg?

I'll see what I can find out about it though...

73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by C0rn3j** [comment]
It's one of the songs from the Westworld soundtracks, no idea how it came to life, but it is definitely not a rip from a video.

**Comment by wagnered** [comment]
Understood...

I tried to pass it through Audacity and export it to a different file
name, tried changing quality then exporting it all to flac format.  The
only way I could get it to play in the web player was to convert it to
high quality (320kb)  mp3.

I know that's not a satisfactory work around.  I did notice that I did
not get a notification when I tried to play your file on the
web-player:  I did on all of my flac files.  This indicates there is
probably a very subtle bug at work, preventing Ampache from transferring
the entire file.

I'll step through and see what I can find, if anything...

73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by wagnered** [comment]
When I tried to debug the problem with Eclipse, the problem doesn't appear.

Maybe some one else has an idea how to track down the problem but right
now, I'm at a loss as how to proceed.

73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by kiv57** [comment]
I switched to debug mode and launched a lossless file that stopped (at 3:54 instead of 5:33) : see ampache log below.

The playback stops more than 1 min after the last log entry.

```
2017-01-28 11:57:16 [admin] (play) -> Playing file (/path/to/file/2006 - The Crane Wife/05 - The Crane Wife - The Perfect Crime #2.m4a}...
2017-01-28 11:57:16 [admin] (play) -> Media type {m4a}
2017-01-28 11:57:16 [admin] (play) -> Custom play action {}
2017-01-28 11:57:16 [admin] (play) -> Transcode to {mp3}
2017-01-28 11:57:16 [admin] (play) -> Transcoding due to explicit request for mp3
2017-01-28 11:57:16 [admin] (stream.class.php) -> Starting transcode for {/path/to/file/2006 - The Crane Wife/05 - The Crane Wife - The Perfect Crime #2.m4a}. Type {mp3}. Options: Array
2017-01-28 11:57:16 [admin] (stream.class.php) -> (
2017-01-28 11:57:16 [admin] (stream.class.php) -> )
2017-01-28 11:57:16 [admin] (stream.class.php) -> }...
2017-01-28 11:57:16 [admin] (media) -> Explicit format request {mp3}
2017-01-28 11:57:16 [admin] (media) -> Transcode settings: from m4a to mp3
2017-01-28 11:57:16 [admin] (media) -> Command: avconv Arguments: -i %FILE% -vn -b:a %SAMPLE%K -c:a libmp3lame -f mp3 pipe:1
2017-01-28 11:57:16 [admin] (stream) -> Active transcoding streams: 0
2017-01-28 11:57:16 [admin] (stream) -> Configured bitrate is 192
2017-01-28 11:57:16 [admin] (stream) -> Final transcode bitrate is 192
2017-01-28 11:57:16 [admin] (stream) -> %BITRATE% not in transcode command
2017-01-28 11:57:16 [admin] (stream) -> %MAXBITRATE% not in transcode command
2017-01-28 11:57:16 [admin] (stream) -> %RESOLUTION% not in transcode command
2017-01-28 11:57:16 [admin] (stream) -> %QUALITY% not in transcode command
2017-01-28 11:57:16 [admin] (stream) -> Transcode command: avconv -i '/path/to/file/2006 - The Crane Wife/05 - The Crane Wife - The Perfect Crime #2.m4a' -vn -b:a 192K -c:a libmp3lame -f mp3 pipe:1
2017-01-28 11:57:16 [admin] (stream) -> Transcode command prefix: exec
2017-01-28 11:57:16 [admin] (play) -> Starting stream of /path/to/file/2006 - The Crane Wife/05 - The Crane Wife - The Perfect Crime #2.m4a with size 39077100
2017-01-28 11:57:16 [admin] (play) -> Registering stream stats for {The Decemberists - The Perfect Crime #2}...
2017-01-28 11:57:16 [admin] (user.class.php) -> Updating stats for {song/14133} {Mozilla/5.0 (Windows NT 6.1; WOW64; rv:50.0) Gecko/20100101 Firefox/50.0}...
2017-01-28 11:57:16 [admin] (user.class.php) -> save_mediaplay...
2017-01-28 11:57:17 [admin] (Last.FM) -> Submission Successful
2017-01-28 11:57:17 [admin] (play) -> Transcode stderr: avconv version 11.7-6:11.7-1~deb8u1, Copyright (c) 2000-2016 the Libav developers
2017-01-28 11:57:17 [admin] (play) ->   built on Jun 12 2016 21:51:35 with gcc 4.9.2 (Debian 4.9.2-10)
2017-01-28 11:57:17 [admin] (play) -> [mov,mp4,m4a,3gp,3g2,mj2 @ 0x945580] max_analyze_duration 5000000 reached
2017-01-28 11:57:17 [admin] (play) -> Input #0, mov,mp4,m4a,3gp,3g2,mj2, from '/path/to/file/2006 - The Crane Wife/05 - The Crane Wife - The Perfect Crime #2.m4a':
2017-01-28 11:57:17 [admin] (play) ->   Metadata:
2017-01-28 11:57:17 [admin] (play) ->     major_brand     : M4A
2017-01-28 11:57:17 [admin] (play) ->     minor_version   : 0
2017-01-28 11:57:17 [admin] (play) ->     compatible_brands: M4A mp42isom
2017-01-28 11:57:17 [admin] (play) ->     creation_time   : 1939-11-17 13:21:14
2017-01-28 11:57:17 [admin] (play) ->     title           : The Perfect Crime #2
2017-01-28 11:57:17 [admin] (play) ->     artist          : The Decemberists
2017-01-28 11:57:17 [admin] (play) ->     album           : The Crane Wife
2017-01-28 11:57:17 [admin] (play) ->     track           : 5/10
2017-01-28 11:57:17 [admin] (play) ->     date            : 2006
2017-01-28 11:57:17 [admin] (play) ->     encoder         : X Lossless Decoder 20110821, QuickTime 7.6.6
2017-01-28 11:57:17 [admin] (play) ->     genre           : Indie Rock
2017-01-28 11:57:17 [admin] (play) ->   Duration: 00:05:33.06, start: 0.000000, bitrate: 938 kb/s
2017-01-28 11:57:17 [admin] (play) ->     Stream #0.0(eng): Audio: alac, 44100 Hz, stereo, s16p, 931 kb/s (default)
2017-01-28 11:57:17 [admin] (play) ->     Metadata:
2017-01-28 11:57:17 [admin] (play) ->       creation_time   : 1939-11-17 13:21:14
2017-01-28 11:57:17 [admin] (play) ->     Stream #0.1: Video: png, rgb24, 600x600, 90k tbn
2017-01-28 11:57:17 [admin] (play) -> Output #0, mp3, to 'pipe:1':
2017-01-28 11:57:17 [admin] (play) ->   Metadata:
2017-01-28 11:57:17 [admin] (play) ->     major_brand     : M4A
2017-01-28 11:57:17 [admin] (play) ->     minor_version   : 0
2017-01-28 11:57:17 [admin] (play) ->     compatible_brands: M4A mp42isom
2017-01-28 11:57:17 [admin] (play) ->     TDEN            : 1939-11-17 13:21:14
2017-01-28 11:57:17 [admin] (play) ->     TIT2            : The Perfect Crime #2
2017-01-28 11:57:17 [admin] (play) ->     TPE1            : The Decemberists
2017-01-28 11:57:17 [admin] (play) ->     TALB            : The Crane Wife
2017-01-28 11:57:17 [admin] (play) ->     TRCK            : 5/10
2017-01-28 11:57:17 [admin] (play) ->     TDRL            : 2006
2017-01-28 11:57:17 [admin] (play) ->     TCON            : Indie Rock
2017-01-28 11:57:17 [admin] (play) ->     TSSE            : Lavf56.1.0
2017-01-28 11:57:17 [admin] (play) ->     Stream #0.0(eng): Audio: libmp3lame, 44100 Hz, stereo, s16p, 192 kb/s (default)
2017-01-28 11:57:17 [admin] (play) ->     Metadata:
2017-01-28 11:57:17 [admin] (play) ->       creation_time   : 1939-11-17 13:21:14
2017-01-28 11:57:17 [admin] (play) ->       encoder         : Lavc56.1.0 libmp3lame
2017-01-28 11:57:17 [admin] (play) -> Stream mapping:
2017-01-28 11:57:17 [admin] (play) ->   Stream #0:0 -> #0:0 (alac (native) -> mp3 (libmp3lame))
2017-01-28 11:57:17 [admin] (play) -> Press ctrl-c to stop encoding
2017-01-28 11:57:17 [admin] (play) ->
2017-01-28 11:57:58 [admin] (session) -> kblliobgmfvqd7kh31j0crnq53 has been extended to Sat, 28 Jan 2017 12:57:58 +0100 extension length 3600
2017-01-28 11:57:58 [admin] (ajax.server.php) -> Called for page: {index}
2017-01-28 11:58:02 [admin] (stream) -> Process is not running, kill skipped.
2017-01-28 11:58:02 [admin] (play) -> Stream ended at 7995023 (7995023) bytes out of
2017-01-28 11:58:57 [admin] (session) -> kblliobgmfvqd7kh31j0crnq53 has been extended to Sat, 28 Jan 2017 12:58:57 +0100 extension length 3600
2017-01-28 11:58:57 [admin] (ajax.server.php) -> Called for page: {index}
```

**Comment by sarcastro** [comment]
I actually had a very similar problem a few weeks ago with a different server-side web-based audio manager... turned out to be an issue with the "silan" silence analyzer version (0.3.2-1build1) packaged with with my distro... I posted a little [how-to fix guide](http://www.twat.ca/how-to-fix-bug-causing-incorrect-song-lengths-in-airtime-2-5-1-running-on-ubuntu-14-04/) when I (eventually) got it figured out, could be worth a check if it's not sorted yet.

**Comment by C0rn3j** [comment]
Silan is not even installed on my system, so unless Ampache does some weird bundling I don't think this is an issue.

**Comment by kiv57** [comment]
Don't have silan installed. Thanks for the tip anyway.

**Comment by kiv57** [comment]
Hi everyone.
I don't know if Ampache is still under development but I still have this issue.
For what it worth it seems to occur only with lossless tracks longer than about 5 min.

Please, just let me know if anyone is working on this, even for the long term, or if this project is declining ?

**Comment by maldata** [comment]
@kiv57 I just started using Ampache, so I just ran into this issue in the last couple days. Some flac files just stop playing mid-song. There didn't appear to be anything different about those files, so I tried re-encoding them, and that seems to work. I encoded the flac file from a CD at the best compression level (8), so decoded it to a wav file and then re-encoded it at level 7. Plays fine now.

It's no more than a workaround, but still, give it a try, I'd be curious if you get the same result.

**Comment by trevhull** [comment]
Don't know if this is the cause of these problems but in case anyone else is looking I was having a similar problem where FLAC transcoded songs would stop before the end of the song when using the web player (but not the android client). The fix I found from  [#135](https://github.com/ampache/ampache/issues/135) was using the flash version of the web player instead of the HTML5. Now all the FLAC transcoded songs play to completion

**Comment by kuzi-moto** [comment]
@C0rn3j was this still an issue for you?

**Comment by C0rn3j** [comment]
I haven't used the software for 3 years, feel free to close it.

**Comment by AshotN** [comment]
That’s a shame to hear you don’t use Ampache anymore. If you ever pick it back up and run into this issue. Feel free to make a new one.
